### ISSUE TITLE
`'NoneType' object has no attribute 'run_command'` when running `terminus_open` and the panel was previously hidden rather than closed

### ISSUE THREAD

**Issue by timfjord** [randy3k/Terminus#322]
### Background

I am working on a sublime plugin where `terminus` acts as a runner.
When I used `terminus_exec` everything was ok, but when I switched to `terminus_open` for more advanced integration I started to get the following when run a command with a **custom panel name** on the second time after the previous panel was hidden(<kbd>Esc</kbd>) rather than closed(<kbd>⌘ + W</kbd>)
```
Traceback (most recent call last):
  File "/Users/timfjord/Library/Application Support/Sublime Text 3/Installed Packages/Terminus.sublime-package/terminus/commands.py", line 96, in <lambda>
  File "/Users/timfjord/Library/Application Support/Sublime Text 3/Installed Packages/Terminus.sublime-package/terminus/commands.py", line 266, in run_async
AttributeError: 'NoneType' object has no attribute 'run_command'
```

When the panel isn't hidden and still in the view or is closed the command runs without problems

### Solution

I was able to fix it by adding a workaround
```python
panel = self.window().find_output_panel(self.PANEL_NAME)
# check if the panel is hidden
if panel is not None and not bool(panel.window()):
  panel.run_command('terminus_close')
```
So basically when the panel is hidden I simply closed it and run the command again

I believe something like this is already present in the `terminus_exec` handler(I just couldn't find it) because there are no issues when doing the same(obviously without the `panel_name` option) with `terminus_exec`

It looks like it has to do with that fact that `window()` returns `None` when the panel is not in the view that's why my workaround is working

---

Just in case here is the implementation https://github.com/timfjord/AnyTest/blob/main/plugin/outputs/terminus.py

**Comment by randy3k** [comment]
It seems to be an issue of a tagged panel. I have pushed a fix and removing the `tag` arg should also fix your issue.

Depends on what you want to do, it may not be necessary to tag a panel. If you launch the command in a differetn wdinwo, it will reuse the tagged panel of the first window.

**Comment by randy3k** [comment]
v0.3.23 is out

**Comment by timfjord** [comment]
Now I understand what you meant that the tags are shared globally. Basically when I want to create a panel with a specific tag and it is already visible but in a different window then my command will be run in that window rather than in my current window.

So I am wondering is it on purpose? I mean are there technical limitations to making that window scoped?
I understand this is a breaking change I am more wondered whether it is possible at all(given the way terminus is implemented)

I think tags should be window specific, I mean why would we want tags to be shared globally? I don't see any use case unless there are some technical limitations.

@randy3k Could you please shed some light on this?

**Comment by randy3k** [comment]
I believe that it should be window specific. It just happened that we have only one dictionary to store the Terminal instances.
Let me think what I could do to addres this.

**Comment by randy3k** [comment]
Fixed. 6f300b9

**Comment by randy3k** [comment]
Ai, the patch doesn't work for panel. Wait for another update.

**Comment by randy3k** [comment]
v0.3.25 is out.

**Comment by timfjord** [comment]
Thanks, @randy3k 

I can confirm tags are now window scoped! :tada:

**Comment by randy3k** [comment]
Just notice that there is another bug preventing this from working correctly. :(.
Creating a new terminal might kill the process of an existing terminal with the same tag.
WIll fix it soon.
